### PR TITLE
full demo w/ 4 metrics

### PR DESCRIPTION
**Pull request by weshayutin**
## Describe the behavior changes introduced in this PR

Takes content from sample apps and consolidates them into the readthedocs demo page.

### CODE REVIEW

**Review by mpryc — CHANGES_REQUESTED** [review]
The following lines do not have target links as ../samples/todolist.md do not exists.

```
* Pelorus is installed and running
  * Follow the steps outlined in the [todolist install guide](../samples/todolist.md#pelorus-install-steps)
* The todolist sample application is installed and running
  * Follow the steps outlined [todolist sample install guide](../samples/todolist.md#todolist-sample-application-install-steps)
  ```

**Comment by weshayutin** [comment]
@mpryc @KevinMGranger ok.. geez.. I think this is ready https://pelorus--540.org.readthedocs.build/en/540/Demo/

File: docs/Demo.md [hunk 1]
```diff
@@ -12,37 +8,137 @@
...
 
-Pelorus should be used as a conversation tool to read the trends in metrics and react by making informed investments in the software delivery process.
+Understand that Pelorus should be used as a conversation tool to read the trends in metrics and react by making informed investments in the software delivery process.
 
-## Prerequisites
+--------------------------
 
-* Pelorus is installed and running
-  * Follow the steps outlined in the [todolist install guide](../samples/todolist.md#pelorus-install-steps)
+# Lead Time for Change and Deployment Frequency
```
Comment (mpryc) [inline docs/Demo.md:21]: I think it's worth mentioning here something like:

Pelorus allows to measure four metrics (https://pelorus.readthedocs.io/en/latest/dashboards/SoftwareDeliveryPerformance/). For this demo two of them will be presented.

File: docs/Demo.md [hunk 2]
```diff
@@ -12,37 +8,137 @@
...
+- install and configure Pelorus
+- install a sample application
 
-```
+### Pelorus install steps
+* Create the pelorus namespace
+
+        oc create namespace pelorus
+
+* Install the required granfa and prometheus operators
```
Comment (mpryc) [inline docs/Demo.md:34]: What about:

```
# Fork pelorus
git clone .... fork
cd fork_dir
helm install ...
```
I think it's better to move the fork / clone to the "prerequisites" section.

File: docs/Demo.md [hunk 3]
```diff
@@ -12,37 +8,137 @@
...
+
+        helm install operators charts/operators --namespace pelorus
+
```
Comment (mpryc) [inline docs/Demo.md:37]: Wait for the grafana-operator-controller-manager and prometheus-operator pods to be Running, similar output to:

```
$ oc get pods -n pelorus
NAME                                                   READY     STATUS    RESTARTS   AGE
grafana-operator-controller-manager-7678cc5c7c-spvls   2/2       Running   0          22s
prometheus-operator-559d659944-fvsjg                   1/1       Running   0          10s

```

File: docs/Demo.md [hunk 4]
```diff
@@ -12,37 +8,137 @@
...
+    
+* Install Pelorus
+    
+        helm install pelorus charts/pelorus --namespace pelorus
+
+* Check the install
+
+        oc get all -n pelorus
+    
+> **Note:** Please pause to allow the install to fully complete.
```
Comment (mpryc) [inline docs/Demo.md:46]: don't think "pause" is valid here... maybe better to check if pods are either Running or Completed state?

```
$ oc get pods -n pelorus
NAME                                                   READY     STATUS      RESTARTS      AGE
deploytime-exporter-1-deploy                           0/1       Completed   0             93s
deploytime-exporter-1-rwk5l                            1/1       Running     0             90s
grafana-deployment-55f77ccc8f-d7m92                    2/2       Running     0             84s
grafana-operator-controller-manager-7678cc5c7c-spvls   2/2       Running     0             4m5s
prometheus-operator-559d659944-fvsjg                   1/1       Running     0             3m53s
prometheus-prometheus-pelorus-0                        3/3       Running     1 (89s ago)   93s
prometheus-prometheus-pelorus-1                        3/3       Running     1 (89s ago)   93s
```

In our e2e script we are checking for the svc services:

```
$ oc get --namespace pelorus svc 
NAME                                                  TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
deploytime-exporter                                   ClusterIP   172.30.178.164   <none>        8080/TCP            4m16s
grafana-operator-controller-manager-metrics-service   ClusterIP   172.30.223.76    <none>        8443/TCP            6m50s
grafana-service                                       ClusterIP   172.30.249.137   <none>        3000/TCP,9091/TCP   4m15s
prometheus-operated                                   ClusterIP   None             <none>        9090/TCP            4m15s
prometheus-pelorus                                    ClusterIP   172.30.90.84     <none>        9090/TCP,9091/TCP   4m16s
```

File: docs/Demo.md [hunk 5]
```diff
@@ -12,37 +8,137 @@
...
+
+### Pelorus configuration
+
+* Make a copy of [the values.yaml file in the pelorus repo](https://github.com/konveyor/pelorus/blob/master/charts/pelorus/values.yaml) ([raw link for curl-ing](https://raw.githubusercontent.com/konveyor/pelorus/master/charts/pelorus/values.yaml)) and save it to /var/tmp/values.yaml
+
+        cp charts/pelorus/values.yaml /var/tmp/
+
+* Update the /var/tmp/values.yaml file with the following example:
```
Comment (mpryc) [inline docs/Demo.md:54]: Update the "exporters" section of the /var/tmp/values.yaml file with the following example:

File: docs/Demo.md [hunk 6]
```diff
@@ -12,37 +8,137 @@
...
+
+        exporters:
```
Comment (mpryc) [inline docs/Demo.md:56]: The following configuartion is exactly the same as yours / removed some helm defaults, so there is no need to specify theAPP_FILE or image_name:

```
exporters:
  instances:
  - app_name: deploytime-exporter
    exporter_type: deploytime
    extraEnv:
    - name: LOG_LEVEL
      value: DEBUG
    - name: NAMESPACES
      value: mongo-persistent
  - app_name: committime-exporter
    exporter_type: committime
    extraEnv:
    - name: LOG_LEVEL
      value: DEBUG
    - name: NAMESPACES
      value: mongo-persistent
```

File: docs/Demo.md [hunk 7]
```diff
@@ -134,49 +229,203 @@ curl $(oc get route -n pelorus deploytime-exporter -o=template='http://{{.spec.h
...
 
 * Please check the logs of exporter pod.
 
 An "idle" state could resemble:
+
 ![Idle-Data](img/pelorus-dashboard-idle-data.png)
 
 
-## Automated Demo
+# Mean Time to Restore and Change Failure Rate
```
Comment (mpryc) [inline docs/Demo.md:249]: Same here, mention that in the previous section two metrics were measured and here another two are presented.

File: docs/Demo.md [hunk 8]
```diff
@@ -134,49 +229,203 @@ curl $(oc get route -n pelorus deploytime-exporter -o=template='http://{{.spec.h
...
+
+### Pelorus configuration
+
+>**Note:**
+> * A users [Github personal access token](https://github.com/settings/tokens) is required
+> * The `PROJECTS` key's value is the fork of the mig-apps-demo repository.
+
+* Update the /var/tmp/values.yaml to match the following text. You may already have the deploytime and committime exporters configured from a previous example.
+
+        exporters:
```
Comment (mpryc) [inline docs/Demo.md:269]: This is exactly same as:
```
exporters:
  instances:
  - app_name: deploytime-exporter
    exporter_type: deploytime
    extraEnv:
    - name: LOG_LEVEL
      value: DEBUG
    - name: NAMESPACES
      value: mongo-persistent
  - app_name: committime-exporter
    exporter_type: committime
    extraEnv:
    - name: LOG_LEVEL
      value: DEBUG
    - name: NAMESPACES
      value: mongo-persistent
  - app_name: failure-exporter
    exporter_type: failure
    extraEnv:
    - name: PROVIDER
      value: github
    - name: TOKEN
      value: ghp_J<snip>
    - name: PROJECTS
      value: <your_org>/mig-demo-apps
```

File: docs/Demo.md [hunk 9]
```diff
@@ -134,49 +229,203 @@ curl $(oc get route -n pelorus deploytime-exporter -o=template='http://{{.spec.h
...
 
-The Pelorus and todolist application can be installed automatically.  Using the forked copy of [mig-demo-apps](https://github.com/konveyor/mig-demo-apps) referenced as `https://github.com/<your_org>/mig-demo-apps`, execute the following steps:
+
+### Pelorus configuration
+
+>**Note:**
+> * A users [Github personal access token](https://github.com/settings/tokens) is required
+> * The `PROJECTS` key's value is the fork of the mig-apps-demo repository.
+
+* Update the /var/tmp/values.yaml to match the following text. You may already have the deploytime and committime exporters configured from a previous example.
```
Comment (mpryc) [inline docs/Demo.md:267]: Update "exporters" section of the /var/tmp/values.yaml to match the following text.

File: docs/Demo.md [hunk 10]
```diff
@@ -134,49 +229,203 @@ curl $(oc get route -n pelorus deploytime-exporter -o=template='http://{{.spec.h
...
+
+* Check the output from the failure exporter.  No bugs should be found at this time.
+
+    
+        curl $(oc get route -n pelorus failure-exporter -o=template='http://{{.spec.host | printf "%s\n"}}')
+    
+
+### Github Issues
+
+Pelorus will utilize two tags to determine if a Github issue is associated with the todolist-mongo application.  We'll need the default `bug` tag. Additionally, Pelorus requires that all issues associated with a particular application be labelled with the `app.kubernetes.io/name=<app_name>` label.  This works the same way as the deployment configuration.
```
Comment (mpryc) [inline docs/Demo.md:322]: Additionally, **by default** Pelorus requires that all issues associated with a particular application be labelled with the `app.kubernetes.io/name=<app_name>` label.  This works the same way as the deployment configuration.

**Review by mpryc — COMMENTED** [review]
Really nice, few small suggestions.

File: docs/Demo.md [hunk 11]
```diff
@@ -1,8 +1,4 @@
 # Pelorus Demo
-## Lead Time for Change and Deployment Frequency
-
-![dora_lead_deployment](img/dora_metrics.png)
-
 
 ### Assumptions
 - oc command line tools installed
```
[the remark this answers is not in the document]
Comment (weshayutin) [reply docs/Demo.md:4]: fixed oc, helm
added:
### Pelorus developer environment

This is an **optional step** for developers that will setup a Python
virtual environment and install prerequisite python and openshift cli tools and libraries.

        cd pelorus
        make dev-env
        source .venv/bin/activate
        make help

File: docs/Demo.md (hunk 1 above)
Comment (weshayutin) [reply docs/Demo.md:21]: Under the DORA img.. I'm adding.
> **Note:** More information about the four key DORA metrics can be found at the [Software Delivery Performance section](Dashboards.md) 

I think that will cover this.

File: docs/Demo.md (hunk 7 above)
Comment (weshayutin) [reply docs/Demo.md:249]: k.. I *think* I added what your looking for, please double check

File: docs/Demo.md (hunk 8 above)
Comment (weshayutin) [reply docs/Demo.md:269]: first one does not include the failure exporter, second one does... should I change that and bring up all three the first time?

File: docs/Demo.md (hunk 10 above)
Comment (weshayutin) [reply docs/Demo.md:322]: updated.. thanks

**Comment by weshayutin** [comment]
for now I've disabled dependbot in the konveyor/mig-demo-apps repo https://github.com/konveyor/mig-demo-apps/settings/security_analysis

**Comment by weshayutin** [comment]
CI was failing due to https://github.com/konveyor/pelorus/issues/545

**Pull request merged**